### PR TITLE
Specialize mongo getresult

### PR DESCRIPTION
**Pull request by lemaitre-aneo**
The factorization of GetResult and GetResults resulted in a very big slow down of Extension Csharp.
This PR specializes GetResult for MongoDB in order to have a faster DB request.

GetResult 17 times faster

### CODE REVIEW

**Review by aneojgurhem — APPROVED** [review]

File: Adaptors/MongoDB/src/ResultTable.cs [hunk 1]
```diff
@@ -94,6 +94,30 @@ public class ResultTable : IResultTable
     }
   }
 
+  async Task<Core.Common.Storage.Result> IResultTable.GetResult(string            sessionId,
+                                                                string            key,
+                                                                CancellationToken cancellationToken)
```
Comment (wkirschenmann) [inline Adaptors/MongoDB/src/ResultTable.cs:99]: =default

File: Adaptors/MongoDB/src/ResultTable.cs [hunk 2]
```diff
@@ -94,6 +94,30 @@ public class ResultTable : IResultTable
...
+  {
+    using var activity = activitySource_.StartActivity($"{nameof(IResultTable.GetResult)}");
```
Comment (wkirschenmann) [inline Adaptors/MongoDB/src/ResultTable.cs:101]: nameof(GetResult)

File: Adaptors/MongoDB/src/ResultTable.cs [hunk 3]
```diff
@@ -94,6 +94,30 @@ public class ResultTable : IResultTable
...
+    activity?.SetTag($"{nameof(IResultTable.GetResult)}_sessionId",
```
Comment (wkirschenmann) [inline Adaptors/MongoDB/src/ResultTable.cs:102]: nameof(GetResult)

File: Adaptors/MongoDB/src/ResultTable.cs [hunk 4]
```diff
@@ -94,6 +94,30 @@ public class ResultTable : IResultTable
...
+                     sessionId);
+    var sessionHandle    = sessionProvider_.Get();
+    var resultCollection = resultCollectionProvider_.Get();
+    try
+    {
+      return await resultCollection.AsQueryable(sessionHandle)
+                                   .Where(model => model.Id == Result.GenerateId(sessionId,
+                                                                                 key))
+                                   .SingleAsync(cancellationToken)
```
Comment (wkirschenmann) [inline Adaptors/MongoDB/src/ResultTable.cs:111]: This request is very different than the one from GetResults (use `model.Id`). Does using this approach in the `GetResults` approach lead to better performances (better usage of indexes) ?

File: Adaptors/MongoDB/src/ResultTable.cs (hunk 1 above)
Comment (aneojgurhem) [reply Adaptors/MongoDB/src/ResultTable.cs:99]: According to R#, not necessary here !

File: Adaptors/MongoDB/src/ResultTable.cs (hunk 2 above)
Comment (aneojgurhem) [reply Adaptors/MongoDB/src/ResultTable.cs:101]: Not working

File: Adaptors/MongoDB/src/ResultTable.cs (hunk 3 above)
Comment (aneojgurhem) [reply Adaptors/MongoDB/src/ResultTable.cs:102]: idem

Comment (wkirschenmann) [reply Adaptors/MongoDB/src/ResultTable.cs:102]: Ok. Strange but OK.

File: Adaptors/MongoDB/src/ResultTable.cs (hunk 1 above)
Comment (wkirschenmann) [reply Adaptors/MongoDB/src/ResultTable.cs:99]: I understand why R# says that it's not necessary. But I'm not sure it's true.
[https://stackoverflow.com/a/8909925](https://stackoverflow.com/a/8909925)

Comment (lemaitre-aneo) [reply Adaptors/MongoDB/src/ResultTable.cs:99]: Optional parameters are given their default value by the caller, not the callee. And the caller does not know what is the actual type of the object, and thus, can only take the default value specified in the base declaration.

File: Adaptors/MongoDB/src/ResultTable.cs (hunk 4 above)
Comment (lemaitre-aneo) [reply Adaptors/MongoDB/src/ResultTable.cs:111]: We can have a look later. But the performance of the batched version seems good, so I'm not worried about that. The goal of this PR was to fix a regression.

**Pull request merged**